confmap/converter/expandconverter/expand.go

```diff
@@ -22,6 +22,7 @@ type converter struct {
 }
 
 // New returns a confmap.Converter, that expands all environment variables for a given confmap.Conf.
+// Deprecated [v0.99.0]: Use NewFactory instead.
```

Comment:
I believe the deprecation notice needs to be on the last line (or on a solo line) for the linter to pick it up. See https://github.com/open-telemetry/opentelemetry-collector/pull/9947